apps/src/lib/node/ledger/shell/mod.rs

```diff
@@ -961,18 +965,32 @@ where
                 );
                 return;
             }
-            if !self.wl_storage.ethbridge_queries().is_bridge_active() {
-                tracing::info!(
-                    "Not starting oracle as the Ethereum bridge is disabled"
-                );
-                return;
-            }
             let Some(config) = EthereumBridgeConfig::read(&self.wl_storage) else {
                 tracing::info!(
                     "Not starting oracle as the Ethereum bridge config couldn't be found in storage"
                 );
                 return;
             };
+            let active =
+                if !self.wl_storage.ethbridge_queries().is_bridge_active() {
+                    if !changed_keys
+                        .contains(&eth_bridge::storage::active_key())
+                    {
+                        tracing::info!(
+                            "Not starting oracle as the Ethereum bridge is \
+                             disabled"
+                        );
+                        return;
+                    } else {
+                        tracing::info!(
+                            "Disabling oracle as the bridge has been disabled"
+                        );
+                        false
+                    }
+                } else {
+                    true
+                };
```

Comment:
Well, we look to see if the key exists so that this function is safe to call before `init_chain` (at which point we don't know if the bridge is active or not). We then ask if the bridge is active (which reads from the write log), I haven't added that. What I added was check if the write log and storage differ, meaning there is an oracle process to talk to. I've never liked that old log line however, and wanted to get rid of it. The new log line is accurate.

Your suggestion is exactly what I'm doing. I just disliked the idea of duplicating logic to check to check these pre-post conditions. It would mean we are doing this computation twice every finalize block call. Once during the application of transactions and once at the tail end when we call this function. So instead I decided to re-use these already computed results.